transaction-scenarios/src/scenarios/non_fungible_resource.rs

```diff
@@ -0,0 +1,405 @@
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    // FIXME: Recalling from frozen vaults should be allowed per product requirement
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-unfreeze-withdraw",
+                        |builder| builder.unfreeze_withdraw(state.vault1.unwrap()),
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-unfreeze-deposit",
+                        |builder| builder.unfreeze_deposit(state.vault1.unwrap()),
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-unfreeze-deposit",
+                        |builder| builder.unfreeze_burn(state.vault1.unwrap()),
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .successful_transaction_with_result_handler(
+                |core, config, state| {
+                    core.next_transaction_with_faucet_lock_fee(
+                        "non-fungible-resource-recall-unfreezed-vault",
+                        |builder| {
+                            builder
+                                .recall_non_fungibles(state.vault1.unwrap(), btreeset!(NonFungibleLocalId::integer(130)))
+                                .try_deposit_batch_or_abort(config.user_account_1.address)
+                        },
+                        vec![&config.user_account_1.key],
+                    )
+                },
+                |core, config, state, result| {
+                    Ok(())
+                },
+            )
+            .finalize(|core, config, state| {
+                Ok(ScenarioOutput {
+                    interesting_addresses: DescribedAddresses::new()
+                    .add("user_account_1", config.user_account_1.address.clone())
```

Comment:
Maybe it's clearer to call this the "main_account" and the other one a "occasional_recipient_account" :')